File: 520/detect-capital.py
class Solution:
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        head, *tail = word
        if head == head.upper():
            return all(list(map(lambda char: char == char.upper(), tail))) or all(list(map(lambda char: char == char.lower(), tail)))

        return all(list(map(lambda char: char == char.lower(), tail)))

File: 520/detect-capital.js
/**
 * @param {string} word
 * @return {boolean}
 */
var detectCapitalUse = function (word) {
    let [head, ...tail] = [...word];
    if (head == head.toUpperCase())
        return tail.every(char => char == char.toUpperCase()) || tail.every(char => char == char.toLowerCase());

    return tail.every(char => char == char.toLowerCase());
};